appjs/views/mixins/actions.js

```diff
@@ -36,6 +36,10 @@ module.exports = {
       $btn.button('loading');
     }
 
+    if(action === 'build-and-publish'){
+      view.$('#projectForm form').submit();
+    }
```

Comment:
Since we are not hiding the publish button after the initial save, and since the user may change the share text on the publish tab, we want to make sure that the project is saved before building.